lib/vsc/utils/fancylogger.py

```diff
@@ -188,12 +192,20 @@ def makeRecord(self, name, level, pathname, lineno, msg, args, excinfo, func=Non
             new_msg = msg.encode('utf8', 'replace')
         return logrecordcls(name, level, pathname, lineno, new_msg, args, excinfo)
 
+    def fail(self, message, *args):
+        """Log error message and raise exception."""
+        formatted_message = message % args
+        self.DEFAULT_RAISE_EXCEPTION_LOG_METHOD(formatted_message)
+        raise self.DEFAULT_RAISE_EXCEPTION_CLASS(formatted_message)
+
     def raiseException(self, message, exception=None, catch=False):
         """
         logs an exception (as warning, since it can be caught higher up and handled)
```

Comment:
maybe change this line to talk about the `RAISE_EXCEPTION_LOG_METHOD`, since warning is gone?